dynamic-rapidpro-workspace-settings/test/index.spec.js

```diff
@@ -0,0 +1,94 @@
+const { expect } = require('chai');
+const path = require('path');
+const sinon = require('sinon');
+const utils = require('../utils');
+const secrets = require('./env');
+const settings = require('./app_settings.json');
+const flows = require('./flows');
+const fs = require('fs');
+const axios = require('axios').default;
+
+let sandbox = sinon.createSandbox();
+
+describe(`rapidpro action test suite`, () => {
+  const mockedAxiosResponse = {
+    data: {},
+    status: 200,
+    statusText: 'OK'
+  };
+  
+  beforeEach(() => {
+    sandbox.stub(process, 'env').value({ 'GITHUB_WORKSPACE': path.join(__dirname, '../') });
+    sandbox.stub(fs, 'writeFileSync').returns({});
+    sandbox.stub(process, 'stdout');
+    sandbox.stub(axios, 'put').resolves(mockedAxiosResponse);
+  });
+
+  afterEach(() => {
+    sandbox.restore();
+  });
+
+  it(`should get a formatted url to set medic-credentials`, async () => {
+    // check the expected url is set
+    const url = utils.getCouchDbUrl(secrets.hostname, secrets.couch_node_name, secrets.value_key, secrets.couch_username, secrets.couch_password);
+    expect(url.origin).to.be.equal(secrets.hostname);
+    expect(url.username).to.be.equal(secrets.couch_username);
+    expect(url.password).to.be.equal(secrets.couch_password);
+  });
+
+  it(`should fail if an invalid url is given`, async () => {
+    expect( function () {
+      utils.getCouchDbUrl('some_invalid_url', secrets.couch_node_name, secrets.value_key, secrets.couch_username, secrets.couch_password);
+    }).to.throw( Error );
+  });
+
+  it(`should return an object containing required secrets`, async () => {
+    const inputs = utils.getInputs(secrets);
+    utils.fields.forEach(field => {
+      expect(inputs[field]).to.equal(secrets[field]);
+    });
+  });
+
+  it(`should fail if no argument is passed to get required secrets`, async () => {
+    expect( function () {
+      utils.getInputs();
```

Comment:
What happens if we pass sone random parameters?

e.g `{}`, `null`, `undefined`

Reply:
Expected to fail, throw `Error`.